styling/dark-mode.md

```diff
@@ -0,0 +1,298 @@
+# Dark Mode in React App
+
+*   [CSS Variables with Context](#css-variables-with-context)
+*   [Styled components](#styled-components)
+*   [MUI](#mui)
+*   [Media Queries](#media-queries)
+
+## CSS Variables with Context
+
+1\. Create a context or state management system to manage the dark mode state across your app. This can be done using React's built-in Context API, Redux, or any other state management library of your choice. The context/state will store a boolean value indicating whether the dark mode is enabled or disabled.
+
+```javascript
+// DarkModeContext.js
+import React, { createContext, useState } from 'react';
```

Comment:
If you use React from version 17 and higher, no need to import `React` from `'react'`
```suggestion
import { createContext, useState } from 'react';
```